pkg/cloud/cloud.go

```diff
@@ -445,3 +446,71 @@ func isBadRequestUpdateInProgress(err error) bool {
 	var badRequest *types.BadRequest
 	return errors.As(err, &badRequest) && strings.Contains(err.Error(), "Unable to perform the storage capacity update. There is an update already in progress.")
 }
+
+func (c *cloud) FindFileSystemByVolumeName(ctx context.Context, volumeName string) (*FileSystem, error) {
+	var nextToken *string
+	const maxResults = 100
+
+	klog.V(4).InfoS("Searching for existing filesystem", "volumeName", volumeName)
+
+	for {
+		input := &fsx.DescribeFileSystemsInput{
```

Comment:
nit: I'd leave a comment explaining that we are doing this pagination/looping since DescribeFileSystems does not support filtering by tags